### PR TITLE
Check response Content-Type

### PR DESCRIPTION
**Pull request by garethsb-sony**
First commit adds checking of the Content-Type header to `GenericTest.check_response`.

Second commit tries to restore some consistency of test helper function naming:
* 'validate' functions raise exceptions
* 'check' functions return a Boolean pass/fail indicator and a message
* 'do_test' functions either return a `TestResult` (or in one case `None`), or raise an `NMOSTestException` when there's an error

### CODE REVIEW

File: GenericTest.py [hunk 1]
```diff
@@ -214,38 +214,59 @@ def prepare_CORS(self, method):
...
+        """Check the Content-Type header of an API request or response"""
+        if "Content-Type" not in headers:
+            return False, "API failed to signal a Content-Type."
+        else:
+            ctype = headers["Content-Type"]
+            ctype_params = ctype.split(";")
+            if ctype_params[0] != "application/json":
+                return False, "API signalled a Content-Type of {} rather than application/json." \
+                              .format(ctype)
+            elif len(ctype_params) == 2 and ctype_params[1].strip() == "charset=utf-8":
```
Comment (andrewbonney) [inline GenericTest.py]: Looks like this may need the same fix as https://github.com/AMWA-TV/nmos-testing/pull/346/files

File: GenericTest.py [hunk 2]
```diff
@@ -214,38 +214,59 @@ def prepare_CORS(self, method):
...
+                return True, "API signalled an unnecessary 'charset' in its Content-Type: {}" \
```
Comment (andrewbonney) [inline GenericTest.py]: The use of 'True' for a warning isn't ideal. It'll do the job, but I suspect it's more likely to trip us up in the future. Would there be any mileage in changing this method to return something like a 'TestState' along with the message? https://github.com/AMWA-TV/nmos-testing/blob/master/TestResult.py#L22

**Comment by garethsb-sony** [comment]
Agreed, the use of `True`-with-a-string to mean WARNING is a little icky.

A lot of testing frameworks have this distinction between checks that cause the test case to be terminated and checks that don't. I think one way to solve this here is a large-ish refactoring that results in not explicitly returning `TestResult` objects from functions but 'setting' them into a 'test context' that is either explicitly passed round or available via `self` e.g. from `GenericTest`. Depending on the `TestState` that was being set, the context would either store the `TestResult` or not (e.g. by default a FAIL or DISABLED would replace a WARNING, but probably a WARNING wouldn't overwrite an existing WARNING, etc.), and then either raise (e.g. for FAIL, DISABLED) or not. The framework would extract the `TestResult` from the context afterwards and if necessary reset it for the next test case.

File: GenericTest.py (hunk 2 above)
Comment (garethsb-sony) [reply GenericTest.py]: Added as #377.

**Pull request merged**